src/eva/vision/data/datasets/segmentation/bcss.py

```diff
@@ -0,0 +1,240 @@
+"""BCSS dataset."""
+
+import glob
+import os
+from pathlib import Path
+from typing import Any, Callable, Dict, List, Literal, Tuple
+
+import numpy as np
+import numpy.typing as npt
+import torch
+from torchvision import tv_tensors
+from torchvision.transforms.v2 import functional
+from typing_extensions import override
+
+from eva.core.data import splitting
+from eva.vision.data.datasets import _validators, wsi
+from eva.vision.data.datasets.segmentation import _utils, base
+from eva.vision.data.wsi.patching import samplers
+from eva.vision.utils import io
+
+
+class BCSS(wsi.MultiWsiDataset, base.ImageSegmentation):
+    """Dataset class for BCSS semantic segmentation task.
+
+    Source: https://github.com/PathologyDataScience/BCSS
+
+    We apply the the class grouping proposed by the challenge baseline:
+    https://bcsegmentation.grand-challenge.org/Baseline/
+
+    outside_roi: outside_roi
+    tumor: angioinvasion, dcis
+    stroma: stroma
+    inflammatory: lymphocytic_infiltrate, plasma_cells, other_immune_infiltrate
+    necrosis: necrosis_or_debris
+    other: remaining
+
+    Be aware that outside_roi should be assigned zero-weight during model training.
+    """
+
+    _train_split_ratio: float = 0.8
+    """Train split ratio."""
+
+    _val_split_ratio: float = 0.2
+    """Validation split ratio."""
+
+    _expected_length: int = 151
+    """Expected dataset length."""
+
+    def __init__(
+        self,
+        root: str,
+        sampler: samplers.Sampler,
+        split: Literal["train", "val", "trainval", "test"] | None = None,
+        width: int = 224,
+        height: int = 224,
+        target_mpp: float = 0.5,
+        transforms: Callable | None = None,
+        seed: int = 42,
```

Comment:
I would prefere to maybe remove the seed from here and fix the dataset on the 42, and we can control the overall seed of eva from only the global one, what do you think?

Reply:
I'll remove the seed & random split here and replace by a static `val` dataset based on a selection of centers as discussed.

Reply:
done